bin/liquid.js

```diff
@@ -1,24 +1,112 @@
+  .option('--layouts <path...>', 'directories from where to resolve layouts (defaults to --root)')
+  .option('--lenient-if', 'do not throw on undefined variables in conditional expressions (when using --strict-variables)')
+  .option('--no-dynamic-paths', 'always treat file paths for partials and layouts as a literal value')
+  .option('--no-greedy-trim', 'disable greedy matching for --trim* options')
+  .option('--no-relative-paths', 'require absolute file paths for partials and layouts')
+  .option('--ordered-filter-parameters', 'respect parameter order when using filters')
+  .option('--output-delimiter-left <string>', 'left delimiter to use for liquid outputs')
+  .option('--output-delimiter-right <string>', 'right delimiter to use for liquid outputs')
+  .option('--partials <path...>', 'directories from where to resolve partials (defaults to --root)')
+  .option('--preserve-timezones', 'preserve input timezone in date filter')
+  .option('--root <path...>', 'directories from where to resolve partials and layouts (defaults to ".")')
+  .option('--strict-filters', 'throw on undefined filters instead of skipping them')
+  .option('--strict-variables', 'throw on undefined variables instead of rendering them as empty string')
+  .option('--tag-delimiter-left', 'left delimiter to use for liquid tags')
+  .option('--tag-delimiter-right', 'right delimiter to use for liquid tags')
+  .option('--timezone-offset <value>', 'JavaScript timezone name or timezoneOffset value to use in date filter (defaults to local timezone)')
+  .option('--trim-output-left', 'trim whitespace from left of liquid outputs')
+  .option('--trim-output-right', 'trim whitespace from right of liquid outputs')
+  .option('--trim-tag-left', 'trim whitespace from left of liquid tags')
+  .option('--trim-tag-right', 'trim whitespace from right of liquid tags')
+  .showHelpAfterError('Use -h or --help for additional information.')
+  .parse()
+
+const options = program.opts()
+const template = resolvePathOption(options.template)
+const context = resolveContext(options.context)
+const liquidOptions = Object.fromEntries(Object.entries(options).map(([key, value]) => {
+  switch (key) { // Remap options where CLI names differ from property names
+    case 'dynamicPaths': return ['dynamicPartials', value]
+    case 'greedyTrim': return ['greedy', value]
+    case 'relativePaths': return ['relativeReference', value]
+    default: return [key, value]
   }
+}))
+
+const liquid = new Liquid(liquidOptions)
+const output = liquid.parseAndRenderSync(template, context)
+if (options.output) {
+  fs.writeFileSync(options.output, output)
+} else {
+  process.stdout.write(output)
+  process.stdout.write('\n')
 }
 
-let tpl = ''
-process.stdin.on('data', chunk => (tpl += chunk))
-process.stdin.on('end', () => render(tpl))
+function resolveContext (contextOption) {
+  let contextJson = '{}'
+  if (contextOption) {
+    contextJson = resolvePathOption(contextOption)
```

Comment:
The previous implementation allowed the context to be specified either as an inline literal or as a path, so I thought it best to retain that feature, as I can see how it might be useful.

However, if you are open to a more robust alternative, I really like the way `curl` does it, for example:

```
curl --data "some inline text data"
curl --data @somedata.txt
echo "some piped text data" | curl --data @-
```

Leading `@` indicates read from file, and `@-` indicates read from `stdin`.

If you like it, we can adopt the same convention here? (But obviously not for legacy mode.)

Reply:
👍🏻 Implemented this, and updating the setup tutorial accordingly.